### PR TITLE
Feature/team member endpoint with CRUD features

### PR DESCRIPTION
**Pull request by trevorkirpaul**
## Description

- implemented model, controllers and routes for TeamMember

## Motivation and Context

- related to issue #37 
## How Has This Been Tested?

- tested through execution, using an mLab instance

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

## Extra

Request body example, used with `create` and `put`

```json
{
	"designation": "testDesignation",
	"name": "Trevor Kirpaul",
	"social_links": {
		"facebook": "blah_facebook",
		"github": "blah_github",
		"linkedin": "blah_linnkedin"
	},
	"profile_pic": "www.google.com"
}
```

That's currently the model, let me know if there needs to be updates! I didn't make anything required and I tried to follow the implementation for other models.

Here are the routes added:

| CRUD |       route         |  type |
|-----------|--------------------|-------- |
| create  | `/team-member/add` | POST |
| delete  |  `/team-member/:id`  | DELETE |
| fetch all | `/team-members`    | GET    |
| fetch one | `/team-member/:id` | GET |
| update | `/team-members/:id` | PUT |

> `create` uses request body shown above 
> `update` uses request body above as well as a URL param being the DB Id

### CODE REVIEW

File: routes/teamMember.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+// Fetch all TeamMembers
+
+router.get('/team-members', TeamMember.getAllTeamMembers);
+
+// delete TeamMember
+
+router.delete('/team-member/:id', TeamMember.deleteTeamMember);
+
+module.exports = router;
```
Comment (outsourcegb) [inline routes/teamMember.js]: 👍 
We can add route for getting single team member. We may use it in future.

Comment (trevorkirpaul) [reply routes/teamMember.js]: Yes, that was something I was wondering about, I'll get that implemented!

**Comment by trevorkirpaul** [comment]
# Added

 - route to fetch a single TeamMember by id

# Updated

- TeamMember controller for fetching a TeamMember by id (`getTeamMember`)

> After testing `getTeamMember` I realized it was broken, I had to refactor it.

File: routes/teamMember.js (hunk 1 above)
Comment (trevorkirpaul) [reply routes/teamMember.js]: Resolved with a commit, refactored the controller method and added a route

File: controllers/team_member.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  teamMemberProfile.findByIdAndUpdate(
+    req.params.id,
+    req.body,
+    { new: true },
+    (err, docs) => {
+      if (err || !docs) {
+        return err
+          ? responseHandler.errpr(res)
+          : responseHandler.error(res, 'TeamMember not found', 404);
+      }
```
[the remark this answers is not in the document]
Comment (Paarmita) [reply controllers/team_member.js]: This might be an 'error' and not 'errpr'.

[the remark this answers is not in the document]
Comment (trevorkirpaul) [reply controllers/team_member.js]: Yup, it's a typo! Fix going up in a a few as I start working on it now.

I'll change the logic to actually make sense as well for returning an error correctly if no Model is found.

**Comment by trevorkirpaul** [comment]
# Updated

- `/controllers/team_member.js` for linting issues, typo and logic with certain methods
- `/config/db.js` to include commented lines referring to a gitignored js file with credentials for connecting mongoose to an mLab instance (or any other db URI)
- `contribution.md` to include the instructions for creating and using the gitignored file (`/config/dev.js`) in order to run the app locally and have it connect to an mLab instance (or any other db URI)
- `gitignore` to include `/config/dev.js`

I tried to get rid of all the linting errors in `/controllers/team_member.js`, I only used on `eslint ignore comment` for the line importing mongoose.

I re-tested all of the routes, including using invalid/non-existing IDs for routes like `update` or `delete`

File: controllers/team_member.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (meyfleur) [reply controllers/team_member.js]: sounds perfect ! thank you :)

**Comment by divyanshu-rawat** [comment]
@trevorkirpaul thanks for the PR 👍

**Pull request merged**